cli/src/test/java/com/devonfw/tools/ide/context/IdeTestContext.java

```diff
@@ -48,6 +52,30 @@ public GitContext getGitContext() {
     return new GitContextMock();
   }
 
+  /**
+   *
+   * @param dummyUserHome mock path which will be used in {@link #getUserHome()}
+   */
+  public void setDummyUserHome(Path dummyUserHome) {
```

Comment:
This is a nice improvement to our test-infrastructure. You could move this up to `AbstractIdeTestContext`.
IMHO it makes more sense to have this symmetic so `setUserHome` sets/changes what `getUserHome()` provides.
```suggestion
  public void setUserHome(Path dummyUserHome) {
```